### PR TITLE
Feature/wx3 support

### PR DESCRIPTION
**Pull request by niffiwan**

### CODE REVIEW

**Comment by niffiwan** [comment]
3rd attempt at the same thing from the 3rd person to attempt it? \:)

I've tested this on Linux & Windows and seems to work OK.

File: src/Main_Panel.cpp [hunk 1]
```diff
@@ -187,16 +187,15 @@ main_panel::main_panel(wxFrame* parent)
 	pgauge = new wxGauge(pstatus, -1, 100,
 								wxPoint(widths[0]+5, 3), 
 								wxSize(widths[1], 17),
-								wxGA_PROGRESSBAR | wxGA_SMOOTH);
+								wxGA_SMOOTH);
```
Comment (z64555) [inline src/Main_Panel.cpp:190]: Why not use wxGA_HORIZONTAL here?

File: src/wxGL_PMFCanvas.h [hunk 2]
```diff
@@ -86,6 +86,8 @@ class wxGL_PMFCanvas : public wxGLCanvas
 
 		void draw_grid(vector3d min, vector3d max, vector3d point);
 
+		wxGLContext* m_context;
```
Comment (z64555) [inline src/wxGL_PMFCanvas.h]: 👍

File: src/Main_Panel.cpp [hunk 3]
```diff
@@ -1192,7 +1192,11 @@ void main_panel::texture_progbar_end(wxAsyncProgressEndEvt &event)
 void main_panel::SignalModelChange(std::string filename, bool skipProgdlg) 
 {
 	UseThreadedProgBar = !skipProgdlg;
-	wxSetWorkingDirectory(wxString(filename.c_str(), wxConvUTF8).BeforeLast('\\'));
+	wxString tdir;
+	wxFileName::SplitPath(wxString(filename.c_str(), wxConvUTF8), &tdir, nullptr, nullptr);
```
Comment (z64555) [inline src/Main_Panel.cpp:1196]: could use a wxDir here?

File: src/MOI.cpp [hunk 4]
```diff
@@ -166,7 +166,7 @@ MOI calc_under_tri(vector3d tri[3], int res){
 		vol_cent.y = plane_line_intersect(pcenter, norm, vol_cent, vector3d(0,1,0), &s).y/2.0f;
 
 		if(s){
-			ret += calc_cuboid_MOI(vol_cent, R-L, abs(vol_cent.y*2), pz);
+			ret += calc_cuboid_MOI(vol_cent, R-L, fabs(vol_cent.y*2), pz);
```
Comment (z64555) [inline src/MOI.cpp:169]: fabs takes in a double and returns a float. Maybe fabsf would be better suited if the arguments are floats?

**Review by z64555 — COMMENTED** [review]
Reviewed what I could. Not going to approve the merge until I can get the thing to build, though.

**Comment by z64555** [comment]
ILU and ILUT are currently in the "Ungrouped" category in CMake, would be good to put those under the same category as IL. Also would be good to rename the IL category to "DevIL"

File: src/wxGL_PMFCanvas.cpp [hunk 5]
```diff
@@ -395,7 +391,7 @@ void wxGL_PMFCanvas::Render()
 		return;
 
 	IsRendering = true;
-	this->SetCurrent();
+	this->SetCurrent(*m_context);
```
Comment (TRBlount) [inline src/wxGL_PMFCanvas.cpp:395]: This should go in wxGL_PMFCanvas::Init() immediately after m_context = new wxGLContext(this);

**Comment by niffiwan** [comment]
frak, I *thought* I'd tested having context set in Init(). 

```
../src/unix/glx11.cpp(86): assert "xid" failed in SetCurrent(): window must be shown
The program 'pcs2' received an X Window System error.
This probably reflects a bug in the program.
The error was 'GLXBadDrawable'.
  (Details: serial 820 error_code 171 request_code 154 minor_code 11)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the --sync command line
   option to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
```

aka X11 doesn't like your solution \:( I'll have a hunt around and see if there's another way of doing this

**Comment by niffiwan** [comment]
ok; thanks to [this page](https://forums.wxwidgets.org/viewtopic.php?t=29948) I've found a different way of doing this which seems to work on Linux.  Now just need to retest on my Windows laptop

**Comment by niffiwan** [comment]
woohoo! This also worked on my Windows laptop. So if @z64555 or @DahBlount could retest & confirm this works we might have a winner (and then I can go & address @z64555's other feedback)

**Comment by TRBlount** [comment]
opengl_init changes compile and run on my windows laptop as well.

**Comment by z64555** [comment]
OK, I'll give the changes a lookover as soon as I get back

**Comment by z64555** [comment]
Confirmed as working. Lets boogie.

**Comment by niffiwan** [comment]
Ah. Thanks! I was intending to squash some commits & address some other comments first, but oh well \:D

File: src/Main_Panel.cpp (hunk 1 above)
Comment (niffiwan) [reply src/Main_Panel.cpp:190]: ok, done

File: src/MOI.cpp (hunk 4 above)
Comment (niffiwan) [reply src/MOI.cpp:169]: ok, done, will probably have a look at all other uses of fabs as well

File: src/Main_Panel.cpp (hunk 3 above)
Comment (niffiwan) [reply src/Main_Panel.cpp:1196]: not sure what the benefit is to using a wxDir? Doesn't seem to be any simpler to use that when getting the dir from a filename & testing that the dir exists.
http://docs.wxwidgets.org/3.0/classwx_dir.html

**Comment by niffiwan** [comment]
@z64555 re this comment:

> ILU and ILUT are currently in the "Ungrouped" category in CMake, would be good to put those under the same category as IL. Also would be good to rename the IL category to "DevIL"

I don't think it's possible, I tried messing around with the variable names in cmake and it breaks builds, probably because `find_package` needs to look for specific names like `IL_LIBRARIES`, `IL_INCLUDE_DIR`, etc.

edit: also grouping seems to occur only by having the same `STRING_` occuring at the start of at least two variables.
https://cmake.org/pipermail/cmake/2009-June/030372.html

**Pull request merged**